subcommand/common/common_test.go

```diff
@@ -26,3 +33,128 @@ func TestValidatePort(t *testing.T) {
 	err = ValidatePort("-test-flag-name", "22")
 	require.EqualError(t, err, "-test-flag-name value of 22 is not in the port range 1024-65535.")
 }
+
+// TestConsulLogin ensures that our implementation of consul login hits `/v1/acl/login`.
+func TestConsulLogin(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	counter := 0
+	client, bearerTokenFile, tokenFile := startMockServer(t, "foo", &counter)
+	err := ConsulLogin(
+		client,
+		bearerTokenFile,
+		testAuthMethod,
+		tokenFile,
+		testPodMeta,
+	)
+	require.NoError(err)
+	require.Equal(counter, 1)
+	// validate that the token file was written to disk
+	data, err := ioutil.ReadFile(tokenFile)
+	require.NoError(err)
+	require.Equal(string(data), "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586")
+}
+
+func TestConsulLogin_EmptyBearerTokenFile(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	counter := 0
+	client, bearerTokenFile, tokenFile := startMockServer(t, "", &counter)
```

Comment:
why do we need the mock server in this test? I think this error happens before any calls to consul are made.

Reply:
Yup, you're right, I was just using the same mock setup for each of the tests, I can be a bit more explicit with the code.